### PR TITLE
ViewModel and UnitTest

### PR DESCRIPTION
**Pull request by nalancer08**
### Type of change

- [ ] Chore
- [X] Feature
- [ ] Bug fix

### Linked issue

- [X] Not tracked

### CODE REVIEW

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                viewModelScope.let { scope ->
+                    val waitFor = scope.async {
+
+                        val workflowResult = getResult {
+                            workflowService.createWorkflow(
+                                postWorkflowBankModel = PostWorkflowBankModel(
+                                    type = PostWorkflowBankModel.Type.plaid,
+                                    kind = PostWorkflowBankModel.Kind.create,
+                                    customerGuid = customerGuid,
+                                    externalBankAccountGuid = "3aa2bc487ae469bbc03cb42cf72e3aa3",
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt:52]: This shouldn't be a cosntant.

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                                    language = PostWorkflowBankModel.Language.en,
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt:53]: This should be read from the settings on the device.

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                                    linkCustomizationName = "default",
+                                    androidPackageName = "app.cybrid.sdkandroid"
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt:55]: These two values should defined in a constants file.

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            if (!cybrid.invalidToken) {
+                viewModelScope.let { scope ->
+                    val waitFor = scope.async {
+
+                        val externalBankAccountResult = getResult {
+                            externalBankAccountsService.createExternalBankAccount(
+                                postExternalBankAccountBankModel = PostExternalBankAccountBankModel(
+                                    name = account?.name ?: "",
+                                    accountKind = PostExternalBankAccountBankModel.AccountKind.plaid,
+                                    asset = "USD",
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt:105]: This needs to be set somehow by the caller. It will depend on the setting of the bank.

File: SDKAndroid/src/test/java/app/cybrid/sdkandroid/components/bankAccounts/BankAccountsViewModelTest.kt [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+        // -- When
+        viewModel.checkWorkflowStatus(workflow)
+
+        // -- Then
+        Assert.assertNotNull(viewModel)
+        Assert.assertNotNull(viewModel.workflowJob)
+        Assert.assertNull(viewModel.latestWorflow)
+        Assert.assertEquals(viewModel.UIState.value, BankAccountsView.BankAccountsViewState.LOADING)
+    }
+}
```
Comment (brent-cybrid) [inline SDKAndroid/src/test/java/app/cybrid/sdkandroid/components/bankAccounts/BankAccountsViewModelTest.kt:190]: These are all positive test cases. You get back many different errors from the various API calls. Check out what you can expect to get back in our OpenAPI spec. It lists them all. We need to test for negative cases as well.

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt (hunk 1 above)
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt:52]: Done!

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt (hunk 2 above)
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt:53]: Done!

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt (hunk 3 above)
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt:55]: Done!

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt (hunk 4 above)
Comment (nalancer08) [reply SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/bankAccounts/view/BankAccountsViewModel.kt]: Done!

File: SDKAndroid/src/test/java/app/cybrid/sdkandroid/components/bankAccounts/BankAccountsViewModelTest.kt (hunk 5 above)
Comment (nalancer08) [reply SDKAndroid/src/test/java/app/cybrid/sdkandroid/components/bankAccounts/BankAccountsViewModelTest.kt:274]: Done!

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/BankAccountsView.kt [hunk 6]
```diff
@@ -0,0 +1,442 @@
...
+        modifier = Modifier
+            .height(120.dp)
+    ) {
+        Column(
+            modifier = Modifier.fillMaxSize(),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            Text(
+                text = "Checking Bank Accounts",
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/BankAccountsView.kt:140]: Needs to be localized

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/BankAccountsView.kt [hunk 7]
```diff
@@ -0,0 +1,442 @@
...
+                    pressedElevation = 4.dp,
+                    disabledElevation = 0.dp
+                ),
+                colors = ButtonDefaults.buttonColors(
+                    backgroundColor = colorResource(id = R.color.primary_color),
+                    contentColor = Color.White
+                )
+            ) {
+                Text(
+                    text = "Add account",
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/BankAccountsView.kt:250]: Localized

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/BankAccountsView.kt [hunk 8]
```diff
@@ -0,0 +1,442 @@
...
+                painter = painterResource(id = R.drawable.kyc_verified),
+                contentDescription = "",
+                modifier = Modifier
+                    .padding(top = 5.dp)
+                    .padding(0.dp)
+                    .size(26.dp),
+                contentScale = ContentScale.Fit
+            )
+            Text(
+                text = "Account added successfully.",
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/BankAccountsView.kt:292]: Localized

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/BankAccountsView.kt [hunk 9]
```diff
@@ -0,0 +1,442 @@
...
+                    pressedElevation = 4.dp,
+                    disabledElevation = 0.dp
+                ),
+                colors = ButtonDefaults.buttonColors(
+                    backgroundColor = colorResource(id = R.color.primary_color),
+                    contentColor = Color.White
+                )
+            ) {
+                Text(
+                    text = "Continue",
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/BankAccountsView.kt:339]: Localized

File: SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/BankAccountsView.kt [hunk 10]
```diff
@@ -0,0 +1,442 @@
...
+                modifier = Modifier
+                    .padding(top = 16.dp),
+                color = colorResource(id = R.color.primary_color)
+            )
+        }
+    }
+}
+
+@Composable
+fun BankAccountsView_Required(viewModel: BankAccountsViewModel?) {
```
Comment (brent-cybrid) [inline SDKAndroid/src/main/java/app/cybrid/sdkandroid/components/BankAccountsView.kt:156]: As mentioned in your previous PR. Can we put each of these Composable functions into their own file so the code is easier to read?

**Review by brent-cybrid — APPROVED** [review]
